src/main/java/models/ShoppingCart.java

```diff
@@ -46,7 +55,7 @@ public Sale checkout() {
                 book.setSale(sale);
             }
             this.removeBooks();
-
+            this.customer.addSale(sale);
```

Comment:
ahhh.. adding this caused the duplication in the dev-UI cuz it was done there as well? If so, it totally makes more sense that it is part of the checkout process (as you've done here)